packages/datetime/src/dateRangePicker.tsx

```diff
@@ -388,9 +389,10 @@ export class DateRangePicker
         if (modifiers.disabled) {
             return;
         }
-        const nextHoverValue = this.getNextValue(this.state.value, day);
-        this.setState({ hoverValue: nextHoverValue });
-        Utils.safeInvoke(this.props.onHoverChange, nextHoverValue, day);
+        const { dateRange, boundary } = DateRangeSelectionStrategy.getNextState(
+            this.state.value, day, this.props.boundaryToModify, this.props.allowSingleDayRange);
```

Comment:
`boundaryToModify` looks suspiciously like it should be part of the strategy mechanics - is it valid?